pkgs/ffigen/lib/src/config_provider/config.dart

```diff
@@ -1,1026 +1,313 @@
-    return [
-      HeterogeneousMapEntry(
-        key: strings.rename,
-        valueConfigSpec: MapConfigSpec<String, dynamic>(
-          schemaDefName: 'rename',
-          keyValueConfigSpecs: [
-            (keyRegexp: '.*', valueConfigSpec: StringConfigSpec()),
-          ],
-        ),
-      ),
-    ];
-  }
-
-  List<HeterogeneousMapEntry> _memberRenameProperties() {
-    return [
-      HeterogeneousMapEntry(
-        key: strings.memberRename,
-        valueConfigSpec: MapConfigSpec<Map<dynamic, String>,
-            Map<dynamic, Map<dynamic, String>>>(
-          schemaDefName: 'memberRename',
-          keyValueConfigSpecs: [
-            (
-              keyRegexp: '.*',
-              valueConfigSpec: MapConfigSpec<String, Map<dynamic, String>>(
-                keyValueConfigSpecs: [
-                  (keyRegexp: '.*', valueConfigSpec: StringConfigSpec())
-                ],
-              ),
-            ),
-          ],
-        ),
-      ),
-    ];
-  }
-
-  List<HeterogeneousMapEntry> _enumIntProperties() => [
-        HeterogeneousMapEntry(
-          key: strings.enumAsInt,
-          defaultValue: (node) => Includer.excludeByDefault(),
-          valueConfigSpec: _includeExcludeObject(),
-        ),
-      ];
-
-  HeterogeneousMapConfigSpec<List<String>, Includer> _includeExcludeObject() {
-    return HeterogeneousMapConfigSpec(
-      schemaDefName: 'includeExclude',
-      entries: [
-        ..._includeExcludeProperties(),
-      ],
-      transform: (node) => extractIncluderFromYaml(node.value),
-    );
-  }
-
-  HeterogeneousMapEntry _dependencyOnlyHeterogeneousMapKey() {
-    return HeterogeneousMapEntry(
-      key: strings.dependencyOnly,
-      valueConfigSpec: EnumConfigSpec<String, CompoundDependencies>(
-        schemaDefName: 'dependencyOnly',
-        allowedValues: {
-          strings.fullCompoundDependencies,
-          strings.opaqueCompoundDependencies,
-        },
-        transform: (node) => node.value == strings.opaqueCompoundDependencies
-            ? CompoundDependencies.opaque
-            : CompoundDependencies.full,
-      ),
-      defaultValue: (node) => CompoundDependencies.full,
-    );
-  }
-
-  MapConfigSpec _mappedTypeObject() {
-    return MapConfigSpec(
-      schemaDefName: 'mappedTypes',
-      keyValueConfigSpecs: [
-        (
-          keyRegexp: '.*',
-          valueConfigSpec: HeterogeneousMapConfigSpec(entries: [
-            HeterogeneousMapEntry(
-                key: strings.lib, valueConfigSpec: StringConfigSpec()),
-            HeterogeneousMapEntry(
-                key: strings.cType, valueConfigSpec: StringConfigSpec()),
-            HeterogeneousMapEntry(
-                key: strings.dartType, valueConfigSpec: StringConfigSpec()),
-          ]),
-        )
-      ],
-      transform: (node) => typeMapExtractor(node.value),
-    );
-  }
+  bool get ignoreSourceErrors;
+
+  /// Whether to format the output file.
+  bool get formatOutput;
+
+  factory Config({
```

Comment:
Can you please rewrite one or two existing yaml config tests to use the `Config(` constructor? (Or duplicate the test so that we have both the yaml and Dart config.) That will show me as reviewer easer what it's usage will look like. (My IDE finds no references.)